app/models/account.rb

```diff
@@ -1,9 +1,19 @@
 # Customer organization account
 class Account < ActiveRecord::Base
+  # @param [String] piece the tenant piece of the canoncial name
+  # @return [String] full canonical name
+  # @see Settings.multitenancy.default_host
+  def self.default_cname(piece)
+    return unless piece
+    default_host = Settings.multitenancy.default_host
+    format(default_host, tenant: piece.parameterize)
+  end
+
   attr_readonly :tenant
-  validates :name, presence: true
+  # name is unused after create, only used by sign_up/new forms
+  validates :name, presence: true, unless: 'cname.present? && cname != default_cname("")'
```

Comment:
Since we have multiple calls to `default_cname` with an empty string passed in, I wonder if we should handle this in that method's definition:

```ruby
def self.default_cname(piece = '')
```

And then call `default_cname` without args in these places?
I'm not sure what it means to call that method with an empty string.